src/bespokelabs/curator/request_processor/base_request_processor.py

```diff
@@ -259,11 +259,19 @@ def create_dataset_files(
                         if prompt_formatter.response_format:
                             # Response message is a string, which is converted to a dict
                             # The dict is then used to construct the response_format Pydantic model
-                            response.response_message = (
-                                prompt_formatter.response_format(
-                                    **response.response_message
+                            try:
+                                response.response_message = (
+                                    prompt_formatter.response_format(
+                                        **response.response_message
+                                    )
                                 )
-                            )
+                            except ValidationError as e:
+                                logger.warning(
+                                    f"Pydantic failed to parse response message {response.response_message} with `response_format` {prompt_formatter.response_format}."
+                                    f"The model likely returned a JSON that does not match the schema of the `response_format`. Will skip this response."
+                                )
+                                response.response_message = None
+                                response.response_errors = [str(e)]
```

Comment:
Small nit: can we add our warning message here too so we can help the user understand the (confusing) pydantic error message